frontend/reducers/task_reducer.js

```diff
@@ -1,31 +1,32 @@
-import { Map, Set } from 'immutable';
+import { Map, Set, fromJS } from 'immutable';
 
 import * as actions from '../consts/action-types';
 
-const defaultState = {
+const defaultState = Map({
   tasksByIds: Map(),
   ids: Set(),
-};
+});
 
 const taskReducer = (state = defaultState, action) => {
   switch (action.type) {
     case actions.RECEIVE_TASKS:
-      return {
-        tasksByIds: state.tasksByIds.merge(action.payload.entities.tasks),
-        ids: state.ids.concat(action.payload.result.tasks),
-      };
+      return state
+        .set('tasksByIds', fromJS(action.payload.entities.tasks))
```

Comment:
payload is js?

Reply:
yes but maybe massageData should convert payload to Immutable.js instead of in reducers?